src/components/OrgFile/components/TaskListModal/index.js

```diff
@@ -0,0 +1,162 @@
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import './stylesheet.css';
+
+import TaskListView from './components/TaskListView';
+import Drawer from '../../../UI/Drawer';
+
+import { isMatch, computeCompletionsForDatalist } from '../../../../lib/headline_filter';
+
+import {
+  extractAllOrgTags,
+  extractAllOrgProperties,
+  extractAllTodoKeywords,
+} from '../../../../lib/org_utils';
+
+import parser from '../../../../lib/headline_filter_parser';
+
+import * as orgActions from '../../../../actions/org';
+
+import _ from 'lodash';
+import format from 'date-fns/format';
+
+class TaskListModal extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    _.bindAll(this, [
+      'handleHeaderClick',
+      'handleToggleDateDisplayType',
+      'handleFilterChange',
+      'handleSearchAllCheckboxChange',
+    ]);
+
+    this.state = {
+      selectedDate: new Date(),
+      dateDisplayType: 'absolute',
+      filterString: '',
+      filterExpr: [],
+      searchAllHeaders: false,
+    };
+  }
+
+  handleHeaderClick(headerId) {
+    this.props.onClose();
+    this.props.org.selectHeaderAndOpenParents(headerId);
+  }
+
+  handleToggleDateDisplayType() {
+    const { dateDisplayType } = this.state;
+
+    this.setState({
+      dateDisplayType: dateDisplayType === 'absolute' ? 'relative' : 'absolute',
+    });
+  }
+
+  handleSearchAllCheckboxChange(event) {
+    const searchAllHeaders = event.target.value;
+    this.setState({ searchAllHeaders });
+  }
+
+  handleFilterChange(event) {
+    const filterString = event.target.value;
+    const curserPosition = event.target.selectionStart;
+    this.setState({ filterString, curserPosition });
+    try {
+      const filterExpr = parser.parse(this.state.filterString);
+      // state change triggers rendering? -> infinite loop
+      this.setState({ filterExpr });
+    } catch (e) {
+      //console.log(e);
+      // TODO: highlight the input (syntax error)
+    }
+  }
+
+  render() {
+    // TODO: is this THE way to get this variable? it must be provided
+    // from parent
+    const { onClose, headers, todoKeywordSets } = this.props;
+    const { selectedDate, dateDisplayType } = this.state;
+
+    let filteredHeaders = headers;
+    if (!_.isEmpty(this.state.filterExpr)) {
+      filteredHeaders = this.props.headers.filter(header => {
+        return isMatch(this.state.filterExpr)(header);
+      });
+    }
+
+    const date = new Date();
+
+    // TODO: read from localStorage (the react way?)
```

Comment:
Should the value of `lastUsedFitlerStrings` be available over multiple usage sessions?

Because if not, after reafactoring this whole logic into Redux, you can read back the info from the store.

And if you want to do it across browser sessions, we still have to go through Redux, so let's first do the this refactor before introducing another side effect with localStorage(;

Reply:
Right, this list should be available over multiple sessions on one device/browser. But each device should have their own list, in my opinion.

Let's wait for the refactoring.

Reply:
I'd say this is part of a bigger initiative, then. This is the current state of offline support of organice is documented here: https://github.com/200ok-ch/organice/#pwa--offline-support

Of course, it would be possible to just include one more piece of information into localStorage (we're already persisting 'settings' and login information [here](https://github.com/200ok-ch/organice/blob/master/src/util/settings_persister.js)). But then we're making the whole system more complicated.

A better approach would be to create a proper architecture for more offline support and then integrate this one feature into this architecture.

Hence, I'd vote to leave this feature out of the current PR (which is already quite big and has many moving parts) and to create a new issue for that with the information from above.

Would that be OK for you?

Reply:
OK, I create the issue.